c/cert/src/rules/EXP36-C/DoNotCastPointerToMoreStrictlyAlignedPointerType.ql

```diff
@@ -0,0 +1,170 @@
+/**
+ * @id c/cert/do-not-cast-pointer-to-more-strictly-aligned-pointer-type
+ * @name EXP36-C: Do not cast pointers into more strictly aligned pointer types
+ * @description Converting a pointer to a different type results in undefined behavior if the
+ *              pointer is not correctly aligned for the new type.
+ * @kind path-problem
+ * @precision high
+ * @problem.severity error
+ * @tags external/cert/id/exp36-c
+ *       correctness
+ *       external/cert/obligation/rule
+ */
+
+import cpp
+import codingstandards.c.cert
+import codingstandards.cpp.Alignment
+import semmle.code.cpp.dataflow.DataFlow
+import semmle.code.cpp.rangeanalysis.SimpleRangeAnalysis
+import DataFlow::PathGraph
+
+/**
+ * An expression with a type that has defined alignment requirements
+ */
+abstract class ExprWithAlignment extends Expr {
+  /**
+   * Gets the alignment requirements in bytes for the underlying `Expr`
+   */
+  abstract int getAlignment();
+
+  /**
+   * Gets a descriptive string describing the type of expression
+   */
+  abstract string getKind();
+}
+
+/**
+ * A class extending `AddressOfExpr` and `ExprWithAlignment` to reason about the
+ * alignment of base types addressed with C address-of expressions
+ */
+class AddressOfAlignedVariableExpr extends AddressOfExpr, ExprWithAlignment {
+  AddressOfAlignedVariableExpr() { this.getAddressable() instanceof Variable }
+
+  AlignAs alignAsAttribute() { result = this.getAddressable().(Variable).getAnAttribute() }
```

Comment:
Should be called `getAlignAsAttribute` to match our standard naming scheme.

Reply:
Changes made.